cooltools/lib/common.py

```diff
@@ -67,3 +72,140 @@ def assign_regions_to_bins(bin_ids, regions_span):
+    """
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset.
+    cache : boolean, optional
+        If True, try to load from the local cache first, and save to the cache
+        if a download is required.
+    data_dir : string, optional
+        The directory where to cache data; default is defined by :func:`get_data_dir`.
+
+    Returns
+    -------
+    Path to the last downloaded file.
+
+    """
+
+    def download_file(url, local_filename=None):
+        if local_filename is None:
+            local_filename = url.split('/')[-1]
+        print('downloading:', url, 'as', local_filename)
+        with requests.get(url, stream=True) as r:
+            r.raise_for_status()
+            with open(local_filename, 'wb') as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+        return local_filename
+
+    available_datasets = _get_datasets_info()
+    available_keys = map(lambda x: x[0], available_datasets)
+
+    if name in available_keys:
+        keys = [name]
+    elif name=="all":
+        keys = available_keys
+    else:
+        raise Exception(
+            """Dataset {key} is not available. 
+            Available datasets: {datasets}
+            Use print_available_datasets() to see the details. 
+            """.format(key=name, datasets=','.join(available_keys)))
+
+    data_dir = get_data_dir(data_dir)
+
+    assert len(keys)>0 # Checking that test data file is parsed successfully
+    downloaded = '' # Empty string that will be returned if the request was empty
+    for key, url, local_filename in available_datasets:
+        if not key in keys:
+            continue
+
+        file_path = os.path.join(data_dir, local_filename)
+        if cache and os.path.exists(file_path):
+            downloaded = file_path
+            continue
+        elif cache:
+            print("Test dataset {} (file {}) is not in the cache directory {}".format(key, local_filename, data_dir))
+        downloaded = download_file(url, file_path)
+
+    return downloaded
+
+
+def get_data_dir(data_dir=None):
+    """
+    Returns a path to cache directory for example datasets.
+
+    This directory is then used by :func:`download_test_data`.
+
+    By default, this function uses the location of cooltools as the home folder
+    and stores the files under ./tests/data/external/ path there.
+
+    Parameters
+    ----------
+    data_dir : str, optional
+        Location of the home for test data storage
+
+    Returns
+    -------
+    String with path to the folder with test data
+
+    Notes
+    -------
+    Designed following seaborn download test data examples:
+    https://github.com/mwaskom/seaborn/blob/4dd57d63e23e4628d8b7f41cdefdf64f7fefcf74/seaborn/utils.py#L427
+
+    """
+
+    if data_dir is None:
+        data_dir = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', '..', 'tests/data/external/'))
```

Comment:
Oh, I just realized. This may not work as expected. The `tests` folder does not "ship" with the package when it is installed the usual way. I think the example data should go somewhere within the package structure, like `cooltools/datasets`.

Reply:
As in sklearn: https://github.com/scikit-learn/scikit-learn/tree/master/sklearn/datasets

Reply:
Good catch! Moved it to `cooltools/datasets`.